pkg/manager/impl/artifact_manager.go

```diff
@@ -60,13 +60,23 @@ func (m *artifactManager) CreateArtifact(ctx context.Context, request datacatalo
 	datasetKey := transformers.FromDatasetID(*artifact.Dataset)
 
 	// The dataset must exist for the artifact, let's verify that first
-	_, err = m.repo.DatasetRepo().Get(ctx, datasetKey)
+	dataset, err := m.repo.DatasetRepo().Get(ctx, datasetKey)
 	if err != nil {
 		logger.Warnf(ctx, "Failed to get dataset for artifact creation %v, err: %v", datasetKey, err)
 		m.systemMetrics.createFailureCounter.Inc(ctx)
 		return nil, err
 	}
 
+	// TODO: when adding a tag, need to verify one tag per partition combo
```

Comment:
When would this be a concern on CreateArtifact? Isn't the tag swapped in a separate step or at least it isn't part of the request currently?

Reply:
That's a good question, this `TODO` may be out of date now because @milohan-lyft requested that tags be allowed to be non-unique within a Dataset and across Partitions.